airframe-control/src/main/scala/wvlet/airframe/control/Retry.scala

```diff
@@ -279,9 +280,14 @@ object Retry extends LogSupport {
         }
 
         resultClass match {
-          case ResultClass.Succeeded =>
+          case ResultClass.Succeeded(x) =>
             // OK. Exit the loop
-            result = Some(ret.get)
+            val clazz = implicitly[ClassTag[A]].runtimeClass
+            if (clazz.isInstance(x)) {
```

Comment:
Using isInstance is not safe if the result is a subclass of A:
https://stackoverflow.com/questions/3949260/java-class-isinstance-vs-class-isassignablefrom

Use isAssingableFrom

Reply:
Fixed.